frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformSection/CreateNewSubformSection.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCard,
+  StudioParagraph,
+  StudioRecommendedNextAction,
+  StudioTextfield,
+} from '@studio/components';
+import { useValidateLayoutSetName } from 'app-shared/hooks/useValidateLayoutSetName';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+import { SubformDataModel } from './SubformDataModel';
+import { CreateNewSubformButtons } from './CreateNewSubformButtons';
+import { SubformInstructions } from './SubformInstructions';
+import { useCreateSubform } from '@altinn/ux-editor/hooks/useCreateSubform';
+
+type CreateNewSubformSectionProps = {
+  layoutSets: LayoutSets;
+  setShowCreateSubformCard: (showCreateSubform: boolean) => void;
+  hasSubforms: boolean;
+  recommendedNextActionText: {
+    title: string;
+    description: string;
+  };
+  onComponentUpdate: (subform: string) => void;
+};
+
+export const CreateNewSubformSection = ({
+  layoutSets,
+  setShowCreateSubformCard,
+  hasSubforms,
+  recommendedNextActionText: { title, description },
+  onComponentUpdate,
+}: CreateNewSubformSectionProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const { validateLayoutSetName } = useValidateLayoutSetName();
+  const [nameError, setNameError] = useState<string | Boolean>(true);
```

Comment:
I'd suggest removing the Boolean type to match the type returned by `validateLayoutSetName`

```suggestion
  const [nameError, setNameError] = useState<string>();
```